generics/services/sunbird.js

```diff
@@ -0,0 +1,211 @@
+            bodyData["request"]["offset"] = offset;
+        }
+        
+          
+        const url = 
+        sunbirdBaseUrl + constants.endpoints.GET_LOCATION_DATA;
+        const options = {
+            headers : {
+                "Authorization" : process.env.SUNBIRD_SERVICE_AUTHERIZATION,
+                "content-type": "application/json"
+            },
+            json : bodyData
+        };
+
+        request.post(url,options,kendraCallback);
+
+        function kendraCallback(err, data) {
+
+            let result = {
+                success : true
+            };
+
+            if (err) {
+                result.success = false;
+            } else {
+                  
+                let response = data.body;
+                  
+                if( response.responseCode === constants.common.OK) {
+                    result["data"] = response.result;
+                } else {
+                      result.success = false;
+                }
+            }
+            return resolve(result);
+        }
+
+        setTimeout(function () {
+           return reject (constants.common.TIMEOUT_ERROR)
+        }, constants.common.SUNBIRD_SERVER_TIMEOUT);
+
+
+      } catch (error) {
+          return reject(error);
+      }
+  })
+}
+
+/**
+  * 
+  * @function
+  * @name formRead
+  * @param {String} bearerToken - autherization token.
+  * @param {object} bodyData -  subType data.
+  * @returns {Promise} returns a promise.
+*/
+const formRead = function ( subTypeData ) {
+    return new Promise(async (resolve, reject) => {
+        try {
+            
+            let bodyData = {
+                request : {
+                    type: "profileConfig",
+                    subType: subTypeData,
+                    action: "get"
+                }
+            }
+            
+            const url = 
+            sunbirdBaseUrl + constants.endpoints.GET_FORM_DATA;
+            const options = {
+                headers : {
+                    "Authorization" : process.env.SUNBIRD_SERVICE_AUTHERIZATION,
+                    "content-type": "application/json"
+                },
+                json : bodyData
+            };
+  
+            request.post(url,options,kendraCallback);
+  
+            function kendraCallback(err, data) {
+  
+                let result = {
+                    success : true
+                };
+  
+                if (err) {
+                    result.success = false;
+                } else {
+                    
+                    let response = data.body;
+                    
+                    if( response.responseCode === constants.common.OK) {
+                        result["data"] = response.result;
+                    } else {
+                        result.success = false;
+                    }
+                }
+                return resolve(result);
+            }
+            setTimeout(function () {
+                return reject (constants.common.TIMEOUT_ERROR)
```

Comment:
@VISHNUDAS-tunerlabs check value of success if its not passing make it false
 if its a api response message add it in generics/api-responses